### PR TITLE
refactor(messages): add text label next to close button - INNO-1047

### PR DESCRIPTION
**Pull request by raduchiriac**
# PR description

List of the complete set of changes this pull request:

#### Dialog
- [refactor] Change `.ecl-message__dismiss` to `.ecl-dialog__dismiss`
```diff
-    <button type="button" class="ecl-message__dismiss" aria-label="{{ _dismiss.label }}">&times</button>
+    <button type="button" class="ecl-dialog__dismiss" aria-label="{{ _dismiss.label }}">&times;</button>
```
- [style] Add Close button outline styling on `:focus`
- [feat] Add class `.ecl-dialog__dismiss--inverted`
- [refactor] Change class that triggers events `.ecl-message__dismiss` 
```diff
-    queryAll('.ecl-message__dismiss').forEach(button => {
+    queryAll('.ecl-dialog__dismiss').forEach(button => {
```

#### Language List
- [refactor] Change of classes `.ecl-dialog__dismiss` and `.ecl-dialog__dismiss--inverted` for the close button. No other class names will close the dialog (see bindings in the [Dialog](https://github.com/ec-europa/europa-component-library/pull/698/files?utf8=%E2%9C%93&diff=unified#diff-86555663e7045413758dfb00a00aa568R140))
```diff
-    <button class="ecl-message__dismiss ecl-message__dismiss--inverted">{{ _messages.dismiss }}</button>
+    <button class="ecl-dialog__dismiss ecl-dialog__dismiss--inverted">{{ _messages.dismiss }}</button>           
```

#### Message
- [feat] Change `.ecl-message`, now displayed as a `display: grid` with a fallback for non-supporting browsers
- [style] Add Close button outline styling on `:focus`
- [feat] Implemented Close button position to the bottom of the container for a mobile version
- [feat] Add new variable `dismiss_text`. `&times;` is no longer the default but `"Close"`
```diff
-    <button type="button" class="ecl-message__dismiss" aria-label="{{ _dismiss_label }}">&times;</button>
+    <button type="button" class="ecl-message__dismiss" aria-label="{{ _dismiss_label }}">{{ _dismiss_text }}</button>
```

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

* [x] `package.json` is up-to-date and `@ecl/[system]-base` is part of the dependencies
* [x] I have checked the dependencies
* [x] I have given the fractal status “ready” to my component
* [x] I have declared `@define mycomponent` in the SCSS file
* [x] I have specified `margin: 0;` on the CSS component
* [x] I have provided tests
* [x] I follow the naming guidelines
* [x] the component supports composition
* [x] there are no hardcoded strings (all content come from the context)
* [x] I have filled the README.md file (at least a few lines)
* [x] My component is listed in the root README
* [x] My PR has the right label(s)

### CODE REVIEW

**Comment by HAL-Patch-INNO** [comment]
Deploy preview for *europa-component-library* ready!

Built with commit 5444932dabad0bc1c71c0efe768113715f027cce

https://deploy-preview-698--europa-component-library.netlify.com

File: src/generic/generic-component-dialog/generic-component-dialog.scss [hunk 1]
```diff
@@ -57,6 +57,26 @@
     width: 100%;
   }
 
+  .ecl-dialog__dismiss {
+    background: transparent url($ecl-assets-path+'images/close.svg') no-repeat
```
Comment (yhuard) [inline src/generic/generic-component-dialog/generic-component-dialog.scss]: missing `@include ecl-focus-outline-border(`

File: src/generic/generic-component-language-list/generic-component-language-list.scss [hunk 2]
```diff
@@ -38,4 +38,50 @@
     box-shadow: none;
     color: #000;
   }
+
+  .ecl-language-list__dismiss {
```
Comment (yhuard) [inline src/generic/generic-component-language-list/generic-component-language-list.scss]: I wouldn't have created a new css class here, only reusing the existing `ecl-dialog__dismiss` and adding the variant to the dialog component too

**Review by yhuard — CHANGES_REQUESTED** [review]
Almost there. Please fix the following bug:

messages___ec_system

File: src/generic/generic-component-message/generic-component-message.scss [hunk 3]
```diff
@@ -41,23 +49,48 @@
...
     border-width: 0;
-    color: transparent;
-    display: block;
-    font-size: map-get($ecl-font-size, 'xxl');
-    line-height: map-get($ecl-font-size, 'l');
+    color: #000;
+    display: flex;
+    font-size: map-get($ecl-font-size, 'xs');
+    height: 1.5rem;
+    line-height: 1.1;
```
Comment (yhuard) [inline src/generic/generic-component-message/generic-component-message.scss]: `1.15` would have been better

File: src/generic/generic-component-message/generic-component-message.scss [hunk 4]
```diff
@@ -41,23 +49,48 @@
...
   .ecl-message__dismiss {
-    background: transparent url($ecl-assets-path+'images/close.svg') no-repeat
-      center center;
+    @include ecl-focus-outline-border(
+      $thicknes: 3px,
+      $color: map-get($ecl-colors, 'yellow-100'),
+      $offset: 3px
+    );
+
+    @supports (display: grid) {
```
Comment (yhuard) [inline src/generic/generic-component-message/generic-component-message.scss]: maybe the code would be more readable if `@supports` comes after the first declarations, before `&:hover {`. Was it stylelint telling you it was wrong?

File: src/systems/ec/ec-component/ec-component-message/ec-component-message.twig [hunk 5]
```diff
@@ -21,8 +21,8 @@
   'messages': messages|default([]),
   'title': title|default(''),
   'dismiss': dismiss|default(false),
-  'dismiss_label': dismiss_label|default(''),
-  'modifier': modifier|default('Dismiss this alert'),
+  'dismiss_label': dismiss_label|default('Close this message'),
+  'modifier': modifier|default(''),
```
Comment (yhuard) [inline src/systems/ec/ec-component/ec-component-message/ec-component-message.twig]: Good catch!

**Review by yhuard — APPROVED** [review]
Overall it looks good and behaves correctly, we're almost ready to merge. 1 small change requested here

File: src/generic/generic-component-message/generic-component-message.scss (hunk 4 above)
Comment (raduchiriac) [reply src/generic/generic-component-message/generic-component-message.scss]: stylelint did not complain, I just thought it makes more sense like this

Comment (yhuard) [reply src/generic/generic-component-message/generic-component-message.scss]: You don't write a modifier before the default class, do you? IMHO it should be the same logic here: default first, then "if feature X is supported"

**Review by yhuard — CHANGES_REQUESTED** [review]
Please update the line-height and the order of the rules (`@supports` after the default) please :)

File: src/generic/generic-component-message/generic-component-message.scss [hunk 6]
```diff
@@ -41,32 +49,67 @@
   }
 
   .ecl-message__title {
+    @supports (display: grid) {
```
Comment (yhuard) [inline src/generic/generic-component-message/generic-component-message.scss]: `@supports` should come after

File: src/generic/generic-component-message/generic-component-message.scss [hunk 7]
```diff
@@ -41,32 +49,67 @@
...
+      grid-column-start: 1;
+    }
+
     font-weight: bold;
     margin-bottom: map-get($ecl-spacing, 'xs');
   }
 
   .ecl-message__body {
+    @supports (display: grid) {
```
Comment (yhuard) [inline src/generic/generic-component-message/generic-component-message.scss]: `@supports` should come after

**Review by yhuard — CHANGES_REQUESTED** [review]
Please provide an upgrade/migration notice, it will help us define if there's a breaking change or not

**Review by degliwe — DISMISSED** [review]

**Review by yhuard — APPROVED** [review]

**Pull request merged**